src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/Languages.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.DotNet.UpgradeAssistant
+{
+    /// <summary>
+    /// An abstraction used to make contextual decisions based on language.
+    /// </summary>
+    public enum Languages
```

Comment:
Nit-pick: Our naming conventions say that enum type names for non-flags enums should be singular so this ought to be `Language`. https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-classes-structs-and-interfaces#naming-enumerations